src/libs/waitForOnyxToClear.js

```diff
@@ -0,0 +1,26 @@
+import createOnReadyTask from './createOnReadyTask';
+
+const onyxClearReadyTask = createOnReadyTask();
+onyxClearReadyTask.setIsReady();
```

Comment:
Can you help me understand why it's being set to ready immediately? It seems to me like this can all be simplified by using the promise returned from `Onyx.clear()`.

Reply:
It is set to be ready immediately because if the transitioning user logged out then we shouldn't wait for Onyx to finish clearing before signing in with the short lived token [here](https://github.com/Expensify/App/blob/bb8bf06226219135493d6580260a6f6ee48396dc/src/pages/LogInWithShortLivedTokenPage.js#L28).

I will refactor this as you suggested in a DM to use promises directly instead of the createOnReadyTask. When I refactor I will initialize waitForOnyxToClear as a resolved promise. When the previous user is signed out I will create a new promise and resolve it when Onyx finishes clearing.